tests/conftest.py

```diff
@@ -0,0 +1,153 @@
+import inspect
```

Comment:
This module should have some docstrings, both at the top of the file, but also for the classes.